src/config/config.cpp

```diff
@@ -0,0 +1,196 @@
+        object.GetValue<std::string>("Algorithm"),
+        object.GetValue<int>("MaxItems"),
+        GetArrayValue<std::string>(
+            object, "ExtendedKeyUsage", [](const Poco::Dynamic::Var& value) { return value.convert<std::string>(); }),
+        GetArrayValue<std::string>(
+            object, "AlternativeNames", [](const Poco::Dynamic::Var& value) { return value.convert<std::string>(); }),
+        object.GetValue<bool>("Disabled"),
+        object.GetValue<bool>("SkipValidation"),
+        object.Get("Params"),
+    };
+}
+
+/***********************************************************************************************************************
+ * Public functions
+ **********************************************************************************************************************/
+
+aos::RetWithError<Config> ParseConfig(const std::string& filename)
+{
+    std::ifstream file(filename);
+
+    if (!file.is_open()) {
+        return {Config {}, aos::ErrorEnum::eNotFound};
+    }
+
+    Config config;
+
+    try {
+        Poco::JSON::Parser           parser;
+        auto                         result = parser.parse(file);
+        CaseInsensitiveObjectWrapper object(result.extract<Poco::JSON::Object::Ptr>());
+
+        config.mIAMPublicServerURL       = object.GetValue<std::string>("IAMPublicServerURL");
+        config.mIAMProtectedServerURL    = object.GetValue<std::string>("IAMProtectedServerURL");
+        config.mNodeID                   = object.GetValue<std::string>("NodeID");
+        config.mNodeType                 = object.GetValue<std::string>("NodeType");
+        config.mCACert                   = object.GetValue<std::string>("CACert");
+        config.mCertStorage              = object.GetValue<std::string>("CertStorage");
+        config.mWorkingDir               = object.GetValue<std::string>("WorkingDir");
+        config.mEnablePermissionsHandler = object.GetValue<bool>("EnablePermissionsHandler");
+
+        config.mFinishProvisioningCmdArgs = GetArrayValue<std::string>(object, "FinishProvisioningCmdArgs",
+            [](const Poco::Dynamic::Var& value) { return value.convert<std::string>(); });
+
+        config.mDiskEncryptionCmdArgs = GetArrayValue<std::string>(object, "DiskEncryptionCmdArgs",
+            [](const Poco::Dynamic::Var& value) { return value.convert<std::string>(); });
+
+        config.mCertModules = GetArrayValue<ModuleConfig>(object, "CertModules", [](const Poco::Dynamic::Var& value) {
+            return ParseModuleConfig(CaseInsensitiveObjectWrapper(value.extract<Poco::JSON::Object::Ptr>()));
+        });
+
+        config.mRemoteIAMs = GetArrayValue<RemoteIAM>(object, "RemoteIAMs", [](const Poco::Dynamic::Var& value) {
+            return ParseRemoteIAM(CaseInsensitiveObjectWrapper(value.extract<Poco::JSON::Object::Ptr>()));
+        });
+
+        config.mIdentifier = ParseIdentifier(object.getObject("Identifier"));
+
+    } catch (const std::exception& e) {
```

Comment:
It's worth logging exception message to simplify troubleshooting.

Reply:
Done